orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	repoFilePermPrivateRW os.FileMode = 0600
+	repoDirPermPrivateRWX os.FileMode = 0700
+)
+
+type Repo struct {
+	fileRepoDir         string
+	fileSuffix          string
+	transientFileMarker string
+
+	mu sync.Mutex
+}
+
+type repoOption func(r *Repo)
+
+func WithTransientFileMarker(marker string) repoOption {
+	return func(r *Repo) {
+		r.transientFileMarker = marker
+	}
+}
```

Comment:
Function pointers are not good for readability IMO. It is good when we want to add hooks (heavily used in linux kernel). I yet to read the updated code again since I reviewed a few days back. Is there no way to remove this?

Reply:
I could change the `New` function to just take an optional variatic `transientFileMarker` list instead and read the first element in that list if it exists to override the default `transientFileMarker`. I was mainly trying to avoid just accepting a `transientFileMarker` argument and ignoring it if it was an empty string since the default would probably be to not pass an override.

Reply:
This is using the OptionFunc pattern for extensible construction. I'm not sure why that's considered difficult to read. It's a flexible, pragmatic, and relatively common way of handling object construction in a friendly, backwards compatible way.

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

The same pattern can be used in place of the "builder" pattern to minimize error handling to one call while building up a bunch of changes.

Reply:
Whenever there is a function pointer and it's usage, I always had to spend some extra time to understand the code. If there is a simple way like `NewServer(config)` or `NewServer(config1, confg2)`, I spend less time to read and understand. This is what I meant regarding the readability. If this is a common pattern and used widely, then I might probably improve my readability skill once I find and read many such patterns.

As I mentioned in my first set of review, I do not see a strong reason to configure the `transientFileMarker`. It is just an internal logic to the filerepo. Given that, I am not convinced that we need a function pointer.